queries.sql

```diff
@@ -16,8 +16,115 @@ SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabu
 
+/* 
+Inside a transaction update the animals table 
+by setting the species column to unspecified. 
+Verify that change was made. Then roll back 
+the change and verify that the species 
+columns went back to 
+the state before the transaction.
+*/
 BEGIN;
 UPDATE animals
 SET species = 'unspecified';
 
-ROLLBACK;
\ No newline at end of file
+ROLLBACK;
+
+/*
+Update the animals table by setting the 
+species column to digimon for all 
+animals that have a name ending in mon.
+*/
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+/*
+Update the animals table by setting 
+the species column to pokemon for all 
+animals that don't have species already set.
+*/
+BEGIN;
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+/*
+Inside a transaction delete all records in the animals table, 
+then roll back the transaction.
+*/
+BEGIN;
+
+DELETE FROM animals;
+ROLLBACK;
+
+-- Delete all animals born after Jan 1st, 2022.
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT DELETEDATE2022;
+
+
+-- Update all animals' weight to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+-- Rollback to the savepoint
+ROLLBACK TO DELETEDATE2022;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+-- Commit transaction
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals
+WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+
+-- Average of escape attempts by neutered
+SELECT neutered, AVG(escape_attempts) FROM animals
+GROUP BY neutered
+HAVING neutered = TRUE;
+
+-- Average of escape attempts by not neutered
+SELECT neutered as NOT_NEUTERED, AVG(escape_attempts) FROM animals
+GROUP BY neutered
+HAVING neutered = FALSE;
+
+-- What is the minimum and maximum weight of each type of animal?
+
+SELECT species, MAX(weight_kg), MIN(weight_kg)
+FROM animals
+GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+
+-- Neutered the average number of escape attempts
+SELECT neutered, AVG(escape_attempts) AS AVG_OF_ESCAPE
+FROM animals
+WHERE neutered = TRUE AND date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY neutered;
+
+-- Not neutered the average number of escape attempts
+SELECT neutered as NOT_NEUTERED, AVG(escape_attempts) AS AVG_OF_ESCAPE
+FROM animals
+WHERE neutered = FALSE AND date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY neutered;
```

Comment:
Hello! 👋. You've got some interesting calculations in there, and it's clear you have a good grasp of using `AVG()` and `GROUP BY`. However, your current code does not meet the specific project requirements. 🛠️

#### Highlights:
- Good use of the `AVG()` function to calculate average values. 🌟
- Your `GROUP BY` and `WHERE` clauses are well-structured. 👍

#### Areas for Improvement:
1. **Wrong Criteria**: The project asks for the average number of escape attempts **per animal type** for animals born between 1990 and 2000. You've calculated it based on whether the animals are neutered or not, which is not what was required.

Here's how your code can meet the requirements:

```sql
SELECT species, AVG(escape_attempts) AS AVG_OF_ESCAPE
FROM animals
WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
GROUP BY species;
```

Please update your query to align with the project requirements and resubmit your code. Can't wait to see your updated work! 🚀

Best,
Temitope